src/benchmark_register.cc

```diff
@@ -159,6 +159,7 @@ bool BenchmarkFamilies::FindBenchmarks(
         instance.use_manual_time = family->use_manual_time_;
         instance.complexity = family->complexity_;
         instance.complexity_lambda = family->complexity_lambda_;
+        instance.statistics = &family->statistics_;
```

Comment:
Hm, correction, https://github.com/google/benchmark/blob/d70417994a3c845c49c4443e92b26a52b320a759/src/benchmark_register.cc#L86
`BenchmarkFamilies::mutex_` does partially help (but no thread safety analysis annotations are used), but it will still not help against the call to `ClearRegisteredBenchmarks()` right after `FindBenchmarks()`. I'm not sure it can happen in reality though.
But i guess `Benchmark::Instance::benchmark` could be a `std::shared_ptr`, and then it would all be fine, with *some* performance penalty.